flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/InputProcessorUtil.java

```diff
@@ -108,18 +108,24 @@ private static CheckpointBarrierHandler createCheckpointBarrierHandler(
 			AbstractInvokable toNotifyOnCheckpoint) {
 		switch (config.getCheckpointMode()) {
 			case EXACTLY_ONCE:
-				if (config.isUnalignedCheckpointsEnabled()) {
-					return new CheckpointBarrierUnaligner(
-						numberOfInputChannelsPerGate.toArray(),
-						channelStateWriter,
+				return config.isUnalignedCheckpointsEnabled() ?
```

Comment:
Again ternary is not making anything easier here.

Reply:
Replied [above](https://github.com/apache/flink/pull/11948#discussion_r418034387).

Reply:
Sorry, I didn't get you point here, can you  please explain
> this change is +/- 0 and doesn't add/remove value

?

(the change is about using `AlternatingCheckpointBarrierHandler`, not ternary instead of if)

Reply:
Yes, the commit makes sense and it has value (using `AlternatingCheckpointBarrierHandler `). 

I meant that changing from if/else to ternary doesn't add or remove much value, it could be argued both ways which option is better, so changing it one way or another in an existing code is a bit pointless, even in separate dedicated commits. Doing it squashed with other changes (especially functional changes) is harmful.

Reply:
Reverted to minimize changes in this PR.